### DISCUSSION TITLE
'@' import not working

### DISCUSSION THREAD

**Discussion by gabriansa** [vercel/next.js#70237] [Help]
### Summary

I get the following error of module not found in my page.jsx: `Module not found: Can't resolve '@/components/ui/button'`.

I am importing with `import { Button } from "@/components/ui/button";` Standard shadcn/ui button.

My `jsconfig.json` is as follows:
```
{
  "compilerOptions": {
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
```

So it should allow for `@` type imports. I can't figure out why I keep getting this error. I tried to create a new project from scratch and still encounter it.

**Comment by mandorakannu** [comment]
You have to pass a baseUrl too. 
```
    "baseUrl": ".",
    "paths": {
      "@/*": ["./*"]
    }
```

**Reply by gabriansa** [reply]
I tried that but I still encounter the same issue. Actually, I noticed that when I run something like `npx shadcn@latest add button` then it creates a new folder `@/components/ui` where it places the component. However it should use the `src`.

**Comment by AmaraFray** [comment]
> ### Summary
> I get the following error of module not found in my page.jsx: `Module not found: Can't resolve '@/components/ui/button'`.
> 
> I am importing with `import { Button } from "@/components/ui/button";` Standard shadcn/ui button.
> 
> My `jsconfig.json` is as follows:
> 
> ```
> {
>   "compilerOptions": {
>     "paths": {
>       "@/*": ["./src/*"]
>     }
>   }
> }
> ```
> 
> So it should allow for `@` type imports. I can't figure out why I keep getting this error. I tried to create a new project from scratch and still encounter it.
> 
> ### Additional information
> _No response_
> 
> ### Example
> _No response_

any solution? when i replace the "@" imports with direct imports like "../" it works. any way to resolve it, our tsconfig is the same

**Comment by pinpos375** [comment]
I encountered the same error. When I tried to install a package from shadcn, it placed it in the root instead of the src directory. Have tried tweaking the jsconfig.json file in all the ways I could think of but it's not working: {
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@/styles/*": ["styles/*"],
      "@/components/*": ["components/*"],
      "@/providers/*": ["providers/*"]
    }
  }
}, Anybody got a solution?

**Comment by shreeram312** [comment]
@gabriansa  @AmaraFray  @mandorakannu  @pinpos375  @toolmantim Did you find out the solution,I am also facing the same problem

**Comment by TirthRadadiya** [comment]
It will depend on what you are using. I am using typescript but I have added extension .jsx or .js file. I renamed extension to .tsx nd .ts. Worked like charm.

**Comment by Mil00Z** [comment]
I have the same issue since yesterday on deploiement flow , and _it's happen on some import,_ not for the all imports elements on project.

I don't understand why and what the hell :/


1.  NextJs 15.1.7, 
2. React 19

+ Typescript

- my `tsconfig.json` is like (the part for Url Path config) :

``` json

"baseUrl": "./",
    "paths": {
      "@/components/*": ["src/components/*"],
      "@/ui/*": ["src/components/ui/*"],
      "@/layout/*": ["src/components/layout/*"],
      "@/styles/*": ["src/styles/*"]
  ```
      
 Errors are like :
 `Module not found: Can't resolve '@/components/ui/Counter/CounterElement'' `

**Reply by mandorakannu** [reply]
Add:
`"@/*": ["src/*"],`
hope this help.

**Reply by Mil00Z** [reply]
THX
I have already this path in my config,(i have shared here a little part of the config).

Yesterday, I found my issue and it's "weird" :
- changing case of files or folder in project (Navigation.jsx to navigation.jsx) doesn't trigger versionning alert or diff between files / folder name.

- So i have diffente name of file between local and repo, and so imports are not correct finaly on the repo.

- When i'm going to trigger the deploiement in Vercel, a lot of error in cascade (not all but one by one) came because the diff betweens imports and file names.

=> I don't understand why this issue with Git / Github and i don't understand why the Log Error in Vercel are not more detailed ^^

**Reply by mandorakannu** [reply]
Reinstall Git and don't change settings

**Comment by JuvitaSaini09** [comment]
In your Next App, If you see two files "**tsconfig.json**" and "**jscofig.json**" then in both of files check is this code present:

1) **tscofig.json**:   "compilerOptions": {
    _"baseUrl": ".",_
    _"paths": {
      "@/*": ["./*"]
    }_,...

2) **jscofig.json**: 
    {
  "compilerOptions": {
    _"baseUrl": ".",_
    _"paths": {
      "@/*": ["./*"]
    }_} }

**Comment by accustraswaraj** [comment]
Are you using `tsx `files in any nested folder inside `/app?` If your project is in jsx, check if a `tsconfig.json` file is getting generated. Deleting that file will make your app run. But you can’t use `tsx `files inside the app folder—it will always generate a `tsconfig.json` and break your app.
I mean, you can use them, but you’d need to do it another way.
